src/games/behavmixed.imp

```diff
@@ -643,130 +589,119 @@ T MixedBehaviorProfile<T>::DiffNodeValue(const GameNode &p_node,
 //             MixedBehaviorProfile<T>: Cached profile information
 //========================================================================
 
+
+// compute realization probabilities for nodes and isets.
+template <class T>
+void MixedBehaviorProfile<T>::ComputePass1_realizProbs(const GameNode &node) const
+{
+  if (node->GetParent()) {
+    map_realizProbs[node] = map_realizProbs[(node->GetParent())] * GetActionProb(node->GetPriorAction());
+  }
+  else {
+    map_realizProbs[node] = (T) 1;
+  }
+
+  if (node->GetInfoset()) {
+    for (auto childNode: node->GetChildren()) {
+      ComputePass1_realizProbs(childNode);
+    }
+  }
+}
+
 template <class T>
-void MixedBehaviorProfile<T>::ComputeSolutionDataPass2(const GameNode &node) const
+void MixedBehaviorProfile<T>::ComputePass2_beliefs_nodeValues_actionValues(const GameNode &node) const
 {
   if (node->GetOutcome()) {
     GameOutcome outcome = node->GetOutcome();
-    for (int pl = 1; pl <= m_support.GetGame()->NumPlayers(); pl++) {
-      m_nodeValues(node->GetNumber(), pl) += static_cast<T>(outcome->GetPayoff(pl));
+    for (auto player: m_support.GetGame()->GetPlayers()) {
+      map_nodeValues[node][player] += static_cast<T>(outcome->GetPayoff(player));
     }
   }
 
   GameInfoset iset = node->GetInfoset();
 
   if (iset) {
     T infosetProb = (T) 0;
-    for (int i = 1; i <= iset->NumMembers(); i++) {
-      infosetProb += m_realizProbs[iset->GetMember(i)->GetNumber()];
+    for (auto i: iset->GetMembers()) {
```

Comment:
I'd suggest `node` or `member` for the loop variable.  However you can also replace this loop with `std::accumulate` to do the sum.

Reply:
Agreed on loop variable.

There's a couple of other places we could use std::accumulate, however, on values of a map like this, I think the code will be less readable -- is a potential speed-up a reason to do this nonetheless? Or is there actually a nice readable implementation with accumulate?